doozerlib/metadata.py

```diff
@@ -300,3 +303,83 @@ def get_maintainer_info(self):
+            if go_sum_file.is_file():
+                try:
+                    # we are looking for a line like: https://github.com/openshift/kubernetes/blob/5241b27b8acd73cdc99a0cac281645189189f1d8/go.sum#L602
+                    # e.g. "k8s.io/kubernetes v1.19.0-rc.2/go.mod h1:zomfQQTZYrQjnakeJi8fHqMNyrDTT6F/MuLaeBHI9Xk="
+                    with go_sum_file.open('r', encoding='utf-8') as f:
+                        for line in f.readlines():
+                            if line.startswith('k8s.io/kubernetes '):
+                                entry_split = line.split()  # => ['k8s.io/kubernetes', 'v1.19.0-rc.2/go.mod', 'h1:zomfQQTZYrQjnakeJi8fHqMNyrDTT6F/MuLaeBHI9Xk=']
+                                base_kube_version = entry_split[1].split('/')[0].strip()  # 'v1.19.0-rc.2/go.mod' => 'v1.19.0-rc.2'
+                                kube_version_fields = base_kube_version.lstrip('v').split('.')  # 'v1.19.0-rc.2' => [ '1', '19', '0-rc.2']
+                                # upstream kubernetes creates a tag for each version. Go find it's sha.
+                                rc, out, err = exectools.cmd_gather('git ls-remote https://github.com/kubernetes/kubernetes {base_kube_version}')
+                                out = out.strip()
+                                if out:
+                                    # Expecting something like 'a26dc584ac121d68a8684741bce0bcba4e2f2957	refs/tags/v1.19.0-rc.2'
+                                    kube_commit_hash = out.split()[0]
+                                else:
+                                    # That's strange, but let's not kill the build for it.  Poke in our repo's hash.
+                                    kube_commit_hash = source_full_sha
+                                break
+                except:
+                    self.logger.error(f'Error parsing go.sum {str(go_sum_file)}')
+                    traceback.print_exc()
+
+            if kube_version_fields:
+                # For historical consistency with tito's flow, we add +OS_GIT_COMMIT[:7] to the kube version
+                envs['KUBE_GIT_VERSION'] = f"v{'.'.join(kube_version_fields)}+{source_full_sha[:7]}"
+                envs['KUBE_GIT_MAJOR'] = '0' if len(kube_version_fields) < 1 else kube_version_fields[0]
+                godep_kube_minor = '0' if len(kube_version_fields) < 2 else kube_version_fields[1]
+                envs['KUBE_GIT_MINOR'] = f'{godep_kube_minor}+'  # For historical reasons, append a '+' since OCP patches its vendored kube.
+                envs['KUBE_GIT_COMMIT'] = kube_commit_hash
+            elif self.name in ('openshift-enterprise-hyperkube', 'openshift'):
+                self.logger.critical('Unable to acquire KUBE vars for openshift-enterprise-hyperkube. This must be fixed or platform addons can break: https://bugzilla.redhat.com/show_bug.cgi?id=1861097')
+                raise IOError('Unable to determine KUBE vars for openshift-enterprise-hyperkube')
```

Comment:
fwiw 3.11 atomic-openshift uses these too.
```suggestion
            elif self.name in ('openshift-enterprise-hyperkube', 'openshift', 'atomic-openshift'):
                self.logger.critical(f'Unable to acquire KUBE vars for {self.name}. This must be fixed or platform addons can break: https://bugzilla.redhat.com/show_bug.cgi?id=1861097')
                raise IOError(f'Unable to determine KUBE vars for {self.name}')
```